pkg/util/iptables/iptables.go

```diff
@@ -601,13 +601,17 @@ func (runner *runner) Monitor(canary Chain, tables []Table, reloadFunc func(), i
 
 // chainExists is used internally by Monitor; none of the public Interface methods can be
 // used to distinguish "chain exists" from "chain does not exist" with no side effects
+// it does not wait to acquire the iptables lock, avoiding to lock the runner unnecessarely
 func (runner *runner) chainExists(table Table, chain Chain) (bool, error) {
+	iptablesCmd := iptablesCommand(runner.protocol)
 	fullArgs := makeFullArgs(table, chain)
+	fullArgs = append([]string{string(opListChain)}, fullArgs...)
 
 	runner.mu.Lock()
 	defer runner.mu.Unlock()
+	klog.V(5).Infof("running iptables: %s %v", iptablesCmd, fullArgs)
+	_, err := runner.exec.Command(iptablesCmd, fullArgs...).CombinedOutput()
```

Comment:
instead of adding an `if op == opListchain` in the `runContext()` method we execute the iptables command directly from here, I think this is cleaner because this `chainExist` method is only used by the monitor

Reply:
Minor, but what about a flag to `run` that said Wait or DontWait ?

Reply:
it turns out this is not the root cause of the problems, not waiting here alleviates the situation because the runner is not locked for 5 seconds but is not causing the lock to be hold forever 
will left that for another time or if someone want to revisit this, seems that Dan has some ideas :smile: 
https://github.com/kubernetes/kubernetes/pull/92860#discussion_r451556669